javaslang/generator/Generator.scala

```diff
@@ -1119,6 +1123,17 @@ def generateMainClasses(): Unit = {
               public Tuple2<T2, T1> swap() {
                 return Tuple.of(_2, _1);
               }
+
+             /$javadoc
+             * Converts the tuple to java.util.Map.Entry {@code Tuple}.
+             *
+             * @return A  java.util.Map.Entry where the first element is the key and the second
+             * element is the value.
+             */
+              public Map.Entry<T1,T2> toEntry(){
+                  return new AbstractMap.SimpleEntry<>(_1, _2);
```

Comment:
@ashrko619 @ruslansennov Let's return a `SimpleEntry`.

Javaslang is immutable-by-design. The `java.util` package is a mutable-by-default parallel universe. The `Map.Entry` has a `setValue()` method. Returning an object that throws would be unsafe.

Thx for asking and giving an alternative. 👏